crates/api/src/publish.rs

```diff
@@ -0,0 +1,131 @@
+//! Kitsune2 publish types.
+
+use crate::{builder, config};
+use crate::{
+    transport::DynTransport, BoxFut, DynFetch, K2Result, OpId, SpaceId, Url,
+};
+use bytes::{Bytes, BytesMut};
+use prost::Message;
+use std::sync::Arc;
+
+pub(crate) mod proto {
+    include!("../proto/gen/kitsune2.publish.rs");
+}
+
+pub use proto::{k2_publish_message::*, K2PublishMessage, PublishOps};
+
+impl From<Vec<OpId>> for PublishOps {
+    fn from(value: Vec<OpId>) -> Self {
+        Self {
+            op_ids: value.into_iter().map(Into::into).collect(),
+        }
+    }
+}
+
+impl From<PublishOps> for Vec<OpId> {
+    fn from(value: PublishOps) -> Self {
+        value.op_ids.into_iter().map(Into::into).collect()
+    }
+}
+
+/// Serialize list of op ids to request.
+pub fn serialize_publish_ops(value: Vec<OpId>) -> Bytes {
```

Comment:
Does this one need to be public? Should be used through the function below?

Reply:
addressed.